client/src/components/Panels/Common/ToolSearch.vue

```diff
@@ -101,20 +110,41 @@ export default {
             return normalizeTools(this.toolbox);
         },
     },
+    mounted() {
+        this.searchWorker = new Worker(new URL("../toolSearch.worker.js", import.meta.url));
```

Comment:
Moving this into a separate module which creates and configures the worker might simplify the usability here. Thanks a lot for working on this. This overall looks really good to me.